### ISSUE TITLE
Unknown crash when running tensorflow.python.ops.gen_nn_ops.max_pool

### ISSUE THREAD

**Issue by dmc1778** [tensorflow/tensorflow#59414]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04

### Python version

3.9

### Current Behaviour?

```shell
A bug happened!
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import os
import numpy as np
from tensorflow.python.ops import gen_nn_ops
try:
  arg_0_tensor = tf.random.uniform([1, 6, 8, 1], dtype=tf.float32)
  arg_0 = tf.identity(arg_0_tensor)
  ksize_0 = 1
  ksize_1 = 3
  ksize_2 = -16
  ksize_3 = 1
  ksize = [ksize_0,ksize_1,ksize_2,ksize_3,]
  strides_0 = 1
  strides_1 = 1
  strides_2 = 1
  strides_3 = 1
  strides = [strides_0,strides_1,strides_2,strides_3,]
  padding = "VALID"
  explicit_paddings = []
  data_format = "NHWC"
  out = gen_nn_ops.max_pool(arg_0,ksize=ksize,strides=strides,padding=padding,explicit_paddings=explicit_paddings,data_format=data_format,)
except Exception as e:
  print("Error:"+str(e))
```
```


### Relevant log output

```shell
2023-01-22 17:52:34.603243: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-22 17:52:34.697556: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-01-22 17:52:35.114243: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/:/home/nimashiri/anaconda3/envs/cuda11.2/lib/:/home/nimashiri/anaconda3/envs/cuda11.2/lib/:/home/nimashiri/anaconda3/envs/tf-2.10/lib/
2023-01-22 17:52:35.114398: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: :/lib/:/home/nimashiri/anaconda3/envs/cuda11.2/lib/:/home/nimashiri/anaconda3/envs/cuda11.2/lib/:/home/nimashiri/anaconda3/envs/tf-2.10/lib/
2023-01-22 17:52:35.114405: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-01-22 17:52:35.589151: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.595319: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.595425: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.595718: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-01-22 17:52:35.596417: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.596518: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.596607: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.944546: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.944690: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.944781: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:980] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2023-01-22 17:52:35.944869: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 4299 MB memory:  -> device: 0, name: NVIDIA GeForce GTX 1660 Ti, pci bus id: 0000:01:00.0, compute capability: 7.5
2023-01-22 17:52:36.501008: I tensorflow/stream_executor/cuda/cuda_dnn.cc:384] Loaded cuDNN version 8100
2023-01-22 17:52:36.501367: F tensorflow/stream_executor/cuda/cuda_dnn.cc:886] Check failed: cudnnSetPoolingNdDescriptor( handle_.get(), (pooling_descriptor.mode() == dnn::PoolingMode::kMaximum ? cudnn_max_pooling_mode : CUDNN_POOLING_AVERAGE_COUNT_EXCLUDE_PADDING), propagate_nans ? CUDNN_PROPAGATE_NAN : CUDNN_NOT_PROPAGATE_NAN, nd, shape.data(), padding.data(), strides.data()) == CUDNN_STATUS_SUCCESS (3 vs. 0)
Aborted

```
```
</details>

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

Apologize for the delay and I was able to replicate the issue without any error when I'm passing correct values to arguments and for your reference I have added [gist-file](https://colab.research.google.com/gist/gaikwadrahul8/2751a2b79ee507efdf04a1b1448e4112/gen_nn_ops-max_pool.ipynb), It seems like you're passing negative value to the kernel/filter/window size and we cannot perform max pooling as you would get a zero or negative dimensions and I think this is the limitation of Max Pooling 

You can refer our [official documentation](https://www.tensorflow.org/api_docs/python/tf/nn/max_pool) and [source-code ](https://github.com/tensorflow/tensorflow/blob/d5b57ca93e506df258271ea00fc29cf98383a374/tensorflow/python/ops/nn_ops.py#L4707-L4842)for High level API `tf.nn.max_pool`

I have replicated on Ubuntu 22.04 with Tensorflow==2.10 and it's working as expected when I'm passing correct arguments with correct values, For your reference I have added screenshot below. 

![image](https://user-images.githubusercontent.com/115997457/215272168-5112ce40-287d-4a6a-bcf6-8ebe0ff84cc3.png)

If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

**Comment by nimashiri** [comment]
> Hi, @nimashiri
> 
> Apologize for the delay and I was able to replicate the issue without any error when I'm passing correct values to arguments and for your reference I have added [gist-file](https://colab.research.google.com/gist/gaikwadrahul8/2751a2b79ee507efdf04a1b1448e4112/gen_nn_ops-max_pool.ipynb), It seems like you're passing negative value to the kernel/filter/window size and we cannot perform max pooling as you would get a zero or negative dimensions and I think this is the limitation of Max Pooling
> 
> You can refer our [official documentation](https://www.tensorflow.org/api_docs/python/tf/nn/max_pool) and [source-code ](https://github.com/tensorflow/tensorflow/blob/d5b57ca93e506df258271ea00fc29cf98383a374/tensorflow/python/ops/nn_ops.py#L4707-L4842)for High level API `tf.nn.max_pool`
> 
> I have replicated on `Ubuntu 22.04` with `Tensorflow==2.10` and it's working as expected when I'm passing correct arguments with correct values, For your reference I have added screenshot below.
> 
> ![image](https://user-images.githubusercontent.com/115997457/215272168-5112ce40-287d-4a6a-bcf6-8ebe0ff84cc3.png)
> 
> If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

Hi, thanks for the reply. If we pass correct values, obviously we wouldn't get any error. I am running a fuzzer on TF APIs (the test cases are machine generated), so I am looking for corner cases. Would you please run my test case?

**Comment by gaikwadrahul8** [comment]
Hi, @nimashiri 

I was able to replicate the issue with your exact same input on `Ubuntu==22.04` with `Tensorflow==2.10` and it's giving me same error message which you're getting because as I mentioned earlier it's due to negative value to the kernel/filter/window size and we cannot perform max pooling as you would get a zero or negative dimensions and this is the limitation of Max Pooling and you can refer [Stackoverflow Answer ](https://stackoverflow.com/questions/49079115/valueerror-negative-dimension-size-caused-by-subtracting-2-from-1-for-max-pool) which will help you to understand the limitations of Max Pooling

I have added screenshot for your reference below 

![image](https://user-images.githubusercontent.com/115997457/215513484-b66de64b-80f9-44b0-b68c-7f5e1a74a385.png)

If issue still persists please let us know? or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!

**Comment by mihaimaruseac** [comment]
Was this tested against nightly?

**Comment by gaikwadrahul8** [comment]
Hi, @mihaimaruseac 

Apologize for the delay and I did not test user code with tf-nightly previously but now I have tested same user code on `Ubuntu 22.04` multiple times with `tf-nightly-2.13.0-dev20230206` and even I tested same user code on Google Colab, here is [gist file](https://colab.research.google.com/gist/gaikwadrahul8/884e47ec7c4920fc0a6854322703c272/-59537-tf-nightly-test.ipynb) and I have added screenshot below. It seems like it's giving expected behaviour. Thank you!

![image](https://user-images.githubusercontent.com/115997457/217207321-1939b906-41a3-4b22-8c17-254cf3d86948.png)
